fix: prevent gitlab app_secret from being sent to the frontend

Halllo5 opened the pull request:
## Summary

The request `/api/connectors/list/` returned the entire GitLab object, including the app secret key. 

I modified the GitLab serializer to remove the secret key. This prevents the key from being sent to the frontend and appears not to break any other functionality

fixes #
prevent gitlab app_secret from being sent to the fronted

<!-- 
### Screenshots (if applicable)

| screenshot 1 | screenshot 2 |
| ------------ | ------------ |
| << 1 >>      | << 2 >>      |
 -->

> [!WARNING]
> Please do not delete the sections below


### ⚠️ If you modify backend code, be sure to run these commands : 

```bash
cd backend
pnpm run openapi # will generate OpenAPI schema at `/openapi/schema.yaml`
pnpm run lock # will update the lockfile if you added a new package
```

### ⚠️ If you modify frontend code, be sure to run these commands : 

```bash
pnpm install --frozen-lockfile # Install the packages at the exact version listed in the lockfile
pnpm run format # format the files using biome
```


## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

Fredkiss3 commented:
hello @Halllo5 , thank you for opening this PR.

> I modified the GitLab serializer to remove the secret key. This prevents the key from being sent to the frontend and appears not to break any other functionality

I breaks something:  The secret can be shown in the edit page of a gitlab app:

<img width="844" height="642" alt="Screenshot 2025-11-08 at 00 08 36" src="https://github.com/user-attachments/assets/3a2d7a8a-4670-4aed-850c-6b946e90053a" />

Now I can understand that you might think it's not secure, but it's not a problem since all the endpoints for ZaneOps require user authentication, so this is not sent publicly.

Halllo5 commented:
Hello @Fredkiss3,
Makes sense, I see your point.
However, I feel that the user doesn't have any reason to retrieve the secret after setting it, as it is not a value that can be set arbitrarily. The only thing that can be done on the GitLab side is regenerate the token completely.

So, a way for the user to update the token makes complete sense, but he doesn't need the old token for that.

IDK, but I don't feel safe deploying it if the secret is sent over the wire this often.  :sweat_smile:

Fredkiss3 commented:
The thing is that, this will break it in a lot of places, not just the serializer or openAPI schema. 

If you try to apply your changes, you will notice that:

1) you would have validation errors because the secret is expected (since the field value will be empty):

<img width="829" height="660" alt="image" src="https://github.com/user-attachments/assets/4cb6e87a-b95e-46ad-94d5-a58581382e12" />

2) And if you remove the secret from the update form serializer, it might break because we expect it from the API  https://github.com/zane-ops/zane-ops/blob/c49b4750fd1c9363613259f71c4c6629e21ded47/backend/git_connectors/views/gitlab.py#L304 
Removing this will make it impossible to update your gitlab app secret. 

---
You'd need to change how the form  for updating the gitlab app is handled on the frontend (and backend side), so no, this is not as easy as it looks.

Halllo5 commented:
@Fredkiss3 
Thanks for your feedback I totally overlooked that :sweat_smile: 
I hope I don't take to much of your time but I think this should about do it.

I've changed the Field to be optional on both front and backed (uses the stored secret if none is provided).
I added an (Optional) mark in the Update screen to make it more clear.

Fredkiss3 commented:
I've approved builds to see if this break tests 🙂
